rest/route/project.go

```diff
@@ -656,6 +656,85 @@ func (h *projectRepotrackerHandler) Run(ctx context.Context) gimlet.Responder {
 	return gimlet.NewJSONResponse(struct{}{})
 }
 
+////////////////////////////////////////////////////////////////////////
+//
+// DELETE /rest/v2/projects/{project_id}
+
+type projectDeleteHandler struct {
+	projectName string
+	sc          data.Connector
+}
+
+func makeDeleteProject(sc data.Connector) gimlet.RouteHandler {
+	return &projectDeleteHandler{
+		sc: sc,
+	}
+}
+
+func (h *projectDeleteHandler) Factory() gimlet.RouteHandler {
+	return &projectDeleteHandler{
+		sc: h.sc,
+	}
+}
+
+func (h *projectDeleteHandler) Parse(ctx context.Context, r *http.Request) error {
+	h.projectName = gimlet.GetVars(r)["project_id"]
+	return nil
+}
+
+func (h *projectDeleteHandler) Run(ctx context.Context) gimlet.Responder {
+	project, err := dbModel.FindOneProjectRef(h.projectName)
+	if err != nil {
+		return gimlet.MakeJSONErrorResponder(err)
+	}
+	if project == nil {
+		return gimlet.MakeJSONErrorResponder(errors.Errorf("project '%s' does not exist", h.projectName))
+	}
+
+	if project.Hidden {
+		return gimlet.MakeJSONErrorResponder(errors.Errorf("project '%s' is already hidden", h.projectName))
+	}
+
+	if !project.UseRepoSettings {
+		return gimlet.MakeJSONErrorResponder(
+			errors.Errorf("project '%s' must have UseRepoSettings enabled to be eligible for deletion", h.projectName))
+	}
+
+	skeletonProj := dbModel.ProjectRef{
+		Id:              project.Id,
+		Owner:           project.Owner,
+		Repo:            project.Repo,
+		Branch:          project.Branch,
+		RepoRefId:       project.RepoRefId,
+		Enabled:         false,
+		UseRepoSettings: true,
+		Hidden:          true,
+	}
+	if updateErr := skeletonProj.Update(); updateErr != nil {
```

Comment:
It's best to just call this err (potentially this will give you a lint "shadowing" error because the variable err had been previously defined; by declaring err here using `=` as opposed to `:=` you're reassigning to err rather than overwriting it, so that will also fix this problem.)

Reply:
Also I didn't catch this earlier, but all of the error returns here should use an errors.Wrap() and specify what is failing (it might not be easy to distinguish from the go driver errors if the projectRef update is failing or the projectVars update is failing, for example)